src/main/joyride/extension.cljs

```diff
@@ -19,7 +20,8 @@
 
 (defn- clear-disposables! []
   (doseq [^js disposable (:disposables @!db)]
-    (.dispose disposable))
+    (p/all
+     (.dispose disposable)))
   (swap! !db assoc :disposables []))

```

Comment:
Should the swap happen after the promise is resolved? Then maybe wrap this in `p/do`?

Reply:
I'll read up on `p/do` to understand the suggestion better.

There are three uses of this that I think off:

1. VS Code gives us a chance to cleanup when the extension is deactivated.
2. We hot reload the extension during our development
3. (Future) Joyride users can re-install whatever VS Code allows in runtime. (This might be something they should do in user space, idk..)

For 1 & 2, I think that what we want here is some best-effort approach, the swap should happen regardless if all disposes succeed. We could do it first, maybe?

Reply:
Just wanted to focus your attention on this, but I trust you will do what makes sense here :)